### PR TITLE
Add DBpedia dataset

### PR DESCRIPTION
**Pull request by dipam7**
## Checklist for adding MMTEB dataset

Reason for dataset addition:

- [x] I have tested that the dataset runs with the `mteb` package.
- [x] I have run the following models on the task (adding the results to the pr). These can be run using the `mteb run -m {model_name} -t {task_name}` command.
  - [x] `sentence-transformers/paraphrase-multilingual-MiniLM-L12-v2`
  - [x] `intfloat/multilingual-e5-small`
- [x] I have checked that the performance is neither trivial (both models gain close to perfect scores) nor random (both models gain close to random scores).
- [x] I have considered the size of the dataset and reduced it if it is too big (2048 examples is typically large enough for most tasks)
- [x] I have filled out the metadata object in the dataset file (find documentation on it [here](https://github.com/embeddings-benchmark/mteb/blob/main/docs/adding_a_dataset.md#2-creating-the-metadata-object)).
- [x] Run tests locally to make sure nothing is broken using `make test`. 
- [x] Run the formatter to format the code using `make lint`. 
- [x] I have added points for my submission to the [points folder](https://github.com/embeddings-benchmark/mteb/blob/main/docs/mmteb/points.md) using the PR number as the filename (e.g. `438.jsonl`).

### CODE REVIEW

File: mteb/tasks/Classification/eng/DBpediaClassification.py [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+        metadata_dict["samples_per_label"] = 4
+        return metadata_dict
+
+    def dataset_transform(self):
+        self.dataset = self.dataset.map(
+            lambda examples: {"text": examples["content"]}, remove_columns=["title"]
+        )
+        self.dataset = self.dataset["test"].train_test_split(
+            test_size=2048, train_size=2048, seed=42, stratify_by_column="label"
+        )
```
Comment (isaac-chung) [inline mteb/tasks/Classification/eng/DBpediaClassification.py:61]: Nice!

**Review by isaac-chung — COMMENTED** [review]
Thanks for adding this! Most of my comments are the same as in https://github.com/embeddings-benchmark/mteb/pull/499

**Comment by dipam7** [comment]
@isaac-chung addressed all the comments

File: docs/mmteb/points/501.jsonl [hunk 2]
```diff
@@ -0,0 +1,2 @@
+{"GitHub": "dipam7", "New dataset": 2}
+{"GitHub": "isaac-chung", "Review PR": 2}
```
Comment (isaac-chung) [inline docs/mmteb/points/501.jsonl:2]:
```suggestion
{"GitHub": "isaac-chung", "Review PR": 2}
{"GitHub": "imenelydiaker", "Review PR": 2}
```

**Review by isaac-chung — APPROVED** [review]
Looks good. Thanks again!

**Pull request merged**